lib/FusionInventory/Agent/Task/Wmi.pm

```diff
@@ -0,0 +1,91 @@
+package FusionInventory::Agent::Task::Wmi;
+
+use strict;
+use warnings;
+use parent 'FusionInventory::Agent::Task::Inventory';
+
+use constant SUPPORTED => map {
+    "FusionInventory::Agent::Task::Inventory::$_"
+} qw(
+        AccessLog
+        Generic
+        Generic::Remote_Mgmt
+        Generic::Remote_Mgmt::TeamViewer
+        Generic::Screen
+        Provider
+        Virtualization
+        Virtualization::HyperV
+        Win32
+        Win32::AntiVirus
+        Win32::Bios
+        Win32::Chassis
+        Win32::Controllers
+        Win32::CPU
+        Win32::Drives
+        Win32::Environment
+        Win32::Firewall
+        Win32::Inputs
+        Win32::License
+        Win32::Memory
+        Win32::Modems
+        Win32::Networks
+        Win32::OS
+        Win32::Ports
+        Win32::Printers
+        Win32::Registry
+        Win32::Slots
+        Win32::Softwares
+        Win32::Sounds
+        Win32::Storages
+        Win32::USB
+        Win32::Users
+        Win32::Videos
+    );
+
+use UNIVERSAL::require;
+use English qw(-no_match_vars);
+
+use FusionInventory::Agent::Tools::Win32;
+
+sub isEnabled {
+    my ($self) = @_;
+
+    # TODO Fix to handle requests from server
+    return defined $self->{service};
+}
+
+sub getModules {
```

Comment:
This change the logic used to select relevant inventory module: the knowledge is in the task, not in each module.

Why not  keep the original logic, and enhance each module isEnabled() function with an additional parameter ? For instance:
package FusionInventory::Agent::Task::Inventory::Generic::Dmidecode;

sub isEnabled {
    my (%params) = @_;
    return 0 if $param->{remote};
    ...
}

Reply:
Here, I think we should keep the isEnabled() for local inventory logic only to keep it as simple as possible.

Maybe we can better introduce a isRemotelyEnabled() API which return false by default and only used in modules able to serve remote inventory ?